src/checkpoint/container/view_traverse_manual.h

```diff
@@ -153,16 +153,9 @@ struct TraverseManual<SerializerT,ViewType,1> {
     using SizeType = typename ViewType::size_type;
     for (SizeType i = 0; i < v.extent(0); i++) {
       if (s.isUnpacking()) {
-        #pragma GCC diagnostic push
-#if !defined(__has_warning)
-        #pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
-#elif __has_warning("-Wmaybe-uninitialized")
-        #pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
-#endif
-        BaseType val;
+        BaseType val{};
```

Comment:
Could this cause problems with constructibility? What types are allowed in Kokkos?